libimagcounter/src/counter.rs

```
@@ -152,12 +154,16 @@ impl<'a> Counter<'a> {
             })
     }
 
-    pub fn unit(&self) -> Option<CounterUnit> {
+    pub fn unit(&self) -> Option<&CounterUnit> {
+        self.unit.as_ref()
+    }
+
+    pub fn read_unit(&self) -> Result<Option<CounterUnit>> {
         self.fle.get_header().read("counter.unit")
-            .ok()
+            .map_err(|e| CEK::StoreReadError.into_error_with_cause(Box::new(e)))
             .and_then(|s| match s {
-                Some(Value::String(s)) => Some(CounterUnit::new(s)),
-                _ => None,
+                Some(Value::String(s)) => Ok(Some(CounterUnit::new(s))),
+                _ => Ok(None),
```

Split it up here:

* In case of `Some(_)` we can return an error indicating that the header is malformed
* In case of `None` we can return `Ok(None)`